tests/unit/services/expects-scoring-test.js

```diff
@@ -74,13 +80,46 @@ module('Unit | Service | expects-scoring', function (hooks) {
   })
 
   test('Only scoreable expects should be part of the results', function (assert) {
-    const nonScoreableExpectation = { id: '1', description: { asSuggestion: 'a suggestion' }, result: true }
-    const e2 = expectation('2', true)
-    const e3 = expectation('3', false)
+    const nonScoreableExpectation = { id: mainTooLongId, description: { asSuggestion: 'a suggestion' }, result: true }
+    const e2 = expectation(doesNotNestControlStructuresId, true)
+    const e3 = expectation(simpleRepetitionId, false)
     const expectations = [nonScoreableExpectation, e2, e3]
 
     const results = expectsScoring.expectsResults(expectations)
 
     assert.propEqual(results, [solutionPassMock, e2, e3])
   })
+
+  const assertFloatEqual = (assert, n1, n2) => {
+    const round = (num) => Number((num).toFixed(3))
+    assert.equal(round(n1), round(n2))
+  }
+
+  test('Total score when using all expects of configuration', function (assert) {
+    const e1 = expectation(doSomethingId, true)
+    const e2 = expectation(tooLongId, false)
+    const e3 = expectation(mainTooLongId, true)
+    const e4 = expectation(nameWasChangedId, true)
+    const e5 = expectation(doesNotNestControlStructuresId, false)
+    const expects = [e1, e2, e3, e4, e5]
+    //The "+1" here refers to the solution passed expectation that is added to the expects.
+    const passingExpects = 3 + 1
+    const totalExpects = 5 + 1
+
+
+    assertFloatEqual(assert, expectsScoring.totalScore(expects, challengeMock), 100 * (passingExpects / totalExpects))
+  })
+
+  test('Calculating total score when not using all expects of the challenge configuration should add the unused expects as passing expects', function (assert) {
+    const e1 = expectation(doSomethingId, true)
+    const e2 = expectation(tooLongId, false)
+    const e3 = expectation(mainTooLongId, false)
+    const e4 = expectation(nameWasChangedId, false)
+    //Does not nest control structures from the decomposition configuration is not used
+    const expects = [e1, e2, e3, e4]
+    const passingExpects = 1 + 2 //The solution passed expectation and the does not nest control structures expectation are added
+    const totalExpects = 5 + 1 //The solution passed expectation is added
+
+    assertFloatEqual(assert, expectsScoring.totalScore(expects, challengeMock), 100 * (passingExpects / totalExpects))
+  })
```

Comment:
:warning: por lo que menciono arriba hay que cambiar el nombre de este test y agregar otro.

- [ ] Si hiciste un programa que no tiene anidación de estructuras de control, el expect no viene desde mulang, así que debería agregarse, y el result por defecto debe ser true y afectar al score (es lo que se testea acá).
- [ ] Si hiciste un programa sin procedimientos ni anidación de estructuras de control, tu score debería ser 2/6 , o sea, sólo se cumplen solutionworks y no anidar. Pero no te doy puntos por main not too long, ni por procedure not too long, doSomething, ni name was changed.

Reply:
Por la forma en la que quedó resuelto siento que son el mismo test, pero de todas formas lo agregué,